libraries/system.rb

```diff
@@ -9,12 +9,12 @@ def initialize(hardware)
 
       def desktop?
         return false if @machine_model.nil?
-        @machine_model.match? Regexp.union %w(Macmini MacPro iMac)
+        @machine_model.match? Regexp.union ['Macmini', 'MacPro', 'iMac', /Mac\d/]
```

Comment:
I feel like there's a better way to determine whether or not a system is laptop or a desktop. Maybe we could check for a built-in display or a battery or both?

Reply:
This does feel a little less elegant now that the regex isn't as clean, but I think the frequency at which new machine models are released is rare enough that this is okay.